ibm_was/datadog_checks/ibm_was/ibm_was.py

```diff
@@ -0,0 +1,141 @@
+                server_tags.extend(node_tags)
+
+                for category, prefix in iteritems(metric_categories):
+                    self.log.debug("Collecting {} stats".format(category))
+                    if collect_stats.get(category):
+                        stats = self.get_node_from_name(server, category)
+                        self.process_stats(stats, prefix, metric_categories, nested_tags, server_tags)
+
+    def get_node_from_name(self, xml_data, path):
+        # XMLPath returns a list, but there should only be one element here since the function starts
+        # the search within a given Node/Server
+        data = xml_data.xpath('//Stat[normalize-space(@name)="{}"]'.format(path))
+        if len(data):
+            return data[0]
+        else:
+            self.warning('Error finding {} stats in XML output.'.format(path))
+            return []
+
+    def get_node_from_root(self, xml_data, path):
+        return xml_data.xpath('//{}'.format(path))
+
+    def process_stats(self, stats, prefix, metric_categories, nested_tags, tags, recursion_level=0):
+        """
+        The XML will have Stat Nodes and Nodes that contain the metrics themselves
+        This code recursively goes through each Stat Node to properly setup tags
+        where each Stat will have a different tag key depending on the context.
+        """
+        for child in stats:
+            if child.tag in metrics.METRIC_VALUE_FIELDS:
+                self.submit_metrics(child, prefix, tags)
+            elif child.tag in metrics.CATEGORY_FIELDS:
+                recursion_tags = tags + ["{}:{}".format(
+                    nested_tags.get(prefix)[recursion_level], child.get('name')
+                )]
+                self.process_stats(child, prefix, metric_categories, nested_tags, recursion_tags, recursion_level+1)
+
+    def submit_metrics(self, child, prefix, tags):
+        value = child.get(metrics.METRIC_VALUE_FIELDS[child.tag])
+        metric_name = self.normalize(
+            ensure_unicode(child.get('name')),
+            prefix='{}.{}'.format(self.METRIC_PREFIX, prefix),
+            fix_case=True
+        )
+        self.metric_type_mapping[child.tag](metric_name, value, tags=tags)
+
+    def make_request(self, instance, url, tags):
+        try:
+            resp = requests.get(url, proxies=self.get_instance_proxy(instance, url))
+            resp.raise_for_status()
+            self.submit_service_checks(tags, AgentCheck.OK)
+        except (requests.HTTPError, requests.ConnectionError) as e:
+            self.warning(
+                "Couldn't connect to URL: {} with exception: {}. Please verify the address is reachable"
+                .format(url, e))
+            self.submit_service_checks(tags, AgentCheck.CRITICAL)
+            raise e
+        return resp.content
+
+    def submit_service_checks(self, tags, value):
+        self.gauge(self.SERVICE_CHECK_CONNECT, value, tags=tags)
```

Comment:
I don't think you should use value directly here, since `OK` is 0, having the metric `can_connect` to 0 intuitively means false IMO. 
So i'd have something like 
```suggestion
        self.gauge(self.SERVICE_CHECK_CONNECT, 1 if value == AgentCheck.OK else 0, tags=tags)
```

Reply:
I just checked the HTTP check and it looks like thats what is currently done there too, will commit this to align with whats there, thanks!

Reply:
I knew I had seen (mb even wrote) it somewhere and didn't remember the check